src/unum/wireless/wireless_nl80211.c

```diff
@@ -0,0 +1,891 @@
+
+    data += 4;
+    len -= 4;
+
+    if (len < 2) {
+        // Too small. Skip
+        buf_offset += snprintf(buf + buf_offset, buf_len - buf_offset,
+                                        ", Pairwise: %s ", defcipher);
+        return;
+    }
+
+    count = data[0] | (data[1] << 8);
+    if (2 + (count * 4) > len) {
+        // Too small. Skip
+        return;
+    }
+
+    // Pairwise Cipher
+    buf_offset += snprintf(buf + buf_offset, buf_len - buf_offset, ", Pairwise: ");
+    for (i = 0; i < count; i++) {
+        if (i != 0) {
+            // Add a space between each supported cipher
+            buf_offset += snprintf(buf + buf_offset, buf_len - buf_offset, " ");
+        }
+        buf_offset += wt_print_cipher(buf + buf_offset, buf_len - buf_offset,
+                                    data + 2 + (i * 4));
+    }
+
+    data += 2 + (count * 4);
+    len -= 2 + (count * 4);
+
+    if (len < 2) {
+        // Too small. Skip
+        buf_offset += snprintf(buf + buf_offset, buf_len - buf_offset,
+                                            ", Auth: %s", defauth);
+        return;
+    }
+
+    count = data[0] | (data[1] << 8);
+    if (2 + (count * 4) > len) {
+        // Too small. Skip
+        return;
+    }
+
+    // Auth algorithm
+    buf_offset += snprintf(buf + buf_offset, buf_len - buf_offset, ", Auth: ");
+    for (i = 0; i < count; i++) {
+        if (i != 0) {
+            // Add a space between each supported cipher
+            buf_offset += snprintf(buf + buf_offset, buf_len - buf_offset, " ");
+        }
+        buf_offset += wt_print_auth(buf + buf_offset, buf_len - buf_offset,
+                                            data + 2 + (i * 4));
+    }
+}
+
+// Print RSN information to buf.
+// Actual processing is done in print_rsn_ie.
+static void wt_print_rsn(char *buf, int buf_len, uint8_t len,
+                    const uint8_t *data)
+{
+    wt_print_rsn_ie(buf, buf_len, "Unknown", "Unknown", len, data);
+}
+
+// Callback on error
+static int wt_error_handler(struct sockaddr_nl *nla,
+                            struct nlmsgerr *err, void *arg)
+{
+    struct nlmsghdr *nlh = (struct nlmsghdr *)err - 1;
+    int len = nlh->nlmsg_len;
+    struct nlattr *attrs;
+    struct nlattr *tb[NLMSGERR_ATTR_MAX + 1];
+    int *ret = arg;
+    int ack_len = sizeof(*nlh) + sizeof(int) + sizeof(*nlh);
+
+    *ret = err->error;
+
+    if (!(nlh->nlmsg_flags & NLM_F_ACK_TLVS))
+        return NL_STOP;
+
+    if (!(nlh->nlmsg_flags & NLM_F_CAPPED))
+        ack_len += err->msg.nlmsg_len - sizeof(*nlh);
+
+    if (len <= ack_len)
+        return NL_STOP;
+
+    attrs = (void *)((unsigned char *)nlh + ack_len);
+    len -= ack_len;
+
+    nla_parse(tb, NLMSGERR_ATTR_MAX, attrs, len, NULL);
+    if (tb[NLMSGERR_ATTR_MSG]) {
+        len = strnlen((char *)nla_data(tb[NLMSGERR_ATTR_MSG]),
+                              nla_len(tb[NLMSGERR_ATTR_MSG]));
+        log("%s: Error: kernel reports: %*s\n", __func__, len,
```

Comment:
Let's print "%s: scan error: %*s\n" instead of "%s: Error: kernel reports: %*s\n".

Reply:
Done.